src/ducks/classrooms.js

```diff
@@ -58,10 +67,54 @@ Effect('getClassrooms', () => {
     .then((data) => {
       Actions.classrooms.setStatus(CLASSROOMS_STATUS.SUCCESS);
       Actions.classrooms.setClassrooms(data);
+      return data;
     }).catch((error) => {
-      Actions.classrooms.setStatus(CLASSROOMS_STATUS.ERROR);
-      Actions.classrooms.setError(error);
-      console.error(error);
+      handleError(error);
+    });
+});
+
+Effect('getClassroomsAndAssignments', () => {
+  Actions.getClassrooms().then((classrooms) => {
+    classrooms.forEach((classroom) => {
+      // TODO: If many pages of assignments exist,
+      // loop through the number of pages to request all of the data
+      // and concatenate the response data together for the app state
+      // Neither Pagination nor infinite scroll would be good UX for current table design.
```

Comment:
In practice, I've never seen a Teacher who so many Classrooms/Assignments that pagination becomes an issue. (Assumption: I2A wouldn't be too different in usage from WildCam Lab.) Personally, I'd put this pretty low on the ToDo/Optimisation list, but if we want to be sure, we can always ask Zach to check the average + max number of Classrooms/Assignments per Teacher so we can better anticipate the data load.